mempool/monitor.go

```diff
@@ -0,0 +1,348 @@
+				p.inventory.Tickets = append([]exptypes.MempoolTx{tx}, p.inventory.Tickets...)
+				p.inventory.NumTickets++
+			case "Vote":
+				// Votes on the next block may be received just prior to dcrdata
+				// actually processing the new block. Do not broadcast these
+				// ahead of the full update with the new block signal as the
+				// vote will be included in that update.
+				if tx.VoteInfo.Validation.Height > p.LastBlockHeight() {
+					p.inventory.Unlock()
+					log.Trace("Got a vote for a future block. Waiting to pull it "+
+						"out of mempool with new block signal. Vote: ", tx.Hash)
+					continue
+				}
+
+				p.inventory.InvStake[tx.Hash] = struct{}{}
+				p.inventory.Votes = append([]exptypes.MempoolTx{tx}, p.inventory.Votes...)
+				//sort.Sort(byHeight(p.inventory.Votes))
+				p.inventory.NumVotes++
+
+				// Multiple transactions can be in mempool from the same ticket.
+				// We need to insure we do not double count these votes.
+				tx.VoteInfo.SetTicketIndex(p.inventory.TicketIndexes)
+				votingInfo := &p.inventory.VotingInfo
+				if tx.VoteInfo.ForLastBlock && !votingInfo.VotedTickets[tx.VoteInfo.TicketSpent] {
+					votingInfo.VotedTickets[tx.VoteInfo.TicketSpent] = true
+					votingInfo.TicketsVoted++
+				}
+			case "Regular":
+				p.inventory.InvRegular[tx.Hash] = struct{}{}
+				p.inventory.Transactions = append([]exptypes.MempoolTx{tx}, p.inventory.Transactions...)
+				p.inventory.NumRegular++
+			case "Revocation":
+				p.inventory.InvStake[tx.Hash] = struct{}{}
+				p.inventory.Revocations = append([]exptypes.MempoolTx{tx}, p.inventory.Revocations...)
+				p.inventory.NumRevokes++
+			}
+
+			// Update latest transactions, popping the oldest transaction off
+			// the back if necessary to limit to NumLatestMempoolTxns.
+			numLatest := len(p.inventory.LatestTransactions)
+			if numLatest >= NumLatestMempoolTxns {
+				p.inventory.LatestTransactions = append([]exptypes.MempoolTx{tx},
+					p.inventory.LatestTransactions[:numLatest-1]...)
+			} else {
+				p.inventory.LatestTransactions = append([]exptypes.MempoolTx{tx},
+					p.inventory.LatestTransactions...)
+			}
+
+			// Store totals.
+			p.inventory.NumAll++
+			p.inventory.TotalOut += tx.TotalOut
+			p.inventory.TotalSize += tx.Size
+			p.inventory.FormattedTotalSize = humanize.Bytes(uint64(p.inventory.TotalSize))
+			p.inventory.Unlock()
+
+			// Broadcast the new transaction.
+			log.Tracef("Signaling mempool event to hub relays...")
+			p.hubSend(pstypes.SigNewTx, time.Second*10)
+
+			log.Tracef("Sending tx data on tx out channels...")
+			p.sendTx(&tx, time.Second*10)
+
+		case <-p.ctx.Done():
+			log.Debugf("Quitting TxHandler (new tx in mempool) handler.")
+			return
+		}
+	}
+}
+
+func (p *MempoolMonitor) hubSend(sig pstypes.HubSignal, timeout time.Duration) {
+	for _, sigout := range p.signalOuts {
+		select {
+		case sigout <- sig:
+		case <-time.After(timeout):
+			log.Errorf("send to signalOuts (%v) failed: Timeout waiting for WebsocketHub.", sig)
```

Comment:
This should not happen during normal operation, so is an Error message.  If this is seen at shutdown, there is an issue with how loops are terminated and/or channels closed, but at least it will not deadlock.